pvn/data/types/base/bolus/calculator/calculator_test.go

```diff
@@ -0,0 +1,305 @@
+				[]*service.Error{testing.SetExpectedErrorSource(validator.ErrorIntegerNotInRange(101, -100, 100), "/recommended/net")},
+			),
+			Entry("correction less than -100", rawObject, "recommended", map[string]interface{}{"net": 50, "correction": -101, "carb": 50},
+				[]*service.Error{testing.SetExpectedErrorSource(validator.ErrorIntegerNotInRange(-101, -100, 100), "/recommended/correction")},
+			),
+			Entry("correction greater than 100", rawObject, "recommended", map[string]interface{}{"net": 50, "correction": 101, "carb": 50},
+				[]*service.Error{testing.SetExpectedErrorSource(validator.ErrorIntegerNotInRange(101, -100, 100), "/recommended/correction")},
+			),
+			Entry("carb less than 0", rawObject, "recommended", map[string]interface{}{"net": 50, "correction": -50, "carb": -1},
+				[]*service.Error{testing.SetExpectedErrorSource(validator.ErrorIntegerNotInRange(-1, 0, 100), "/recommended/carb")},
+			),
+			Entry("carb greater than 100", rawObject, "recommended", map[string]interface{}{"net": 50, "correction": -50, "carb": 101},
+				[]*service.Error{testing.SetExpectedErrorSource(validator.ErrorIntegerNotInRange(101, 0, 100), "/recommended/carb")},
+			),
+		)
+
+		DescribeTable("valid when", testing.ExpectFieldIsValid,
+			Entry("net more or equal -100", rawObject, "recommended", map[string]interface{}{"net": -100, "correction": -50, "carb": 50}),
+			Entry("net less or equal 100", rawObject, "recommended", map[string]interface{}{"net": 100, "correction": -50, "carb": 50}),
+			Entry("correction more or equal -100", rawObject, "recommended", map[string]interface{}{"net": 10, "correction": -100, "carb": 50}),
+			Entry("correction less or equal 100", rawObject, "recommended", map[string]interface{}{"net": 10, "correction": 100, "carb": 50}),
+			Entry("carb more or equal 0", rawObject, "recommended", map[string]interface{}{"net": -100, "correction": -50, "carb": 0}),
+			Entry("carb less or equal 50", rawObject, "recommended", map[string]interface{}{"net": 100, "correction": -50, "carb": 50}),
+		)
+
+	})
+
+	Context("Normalize", func() {
+
+		Context("blood glucose", func() {
+
+			DescribeTable("when mmol/L", func(val, expected float64) {
+				bolusCalculator := calculator.New()
+				bolusCalculator.Units = &common.Mmoll
+				bolusCalculator.BloodGlucoseInput = &val
+				bolusCalculator.InsulinSensitivity = &val
+				bolusCalculator.BloodGlucoseTarget = &calculator.BloodGlucoseTarget{Target: &val}
+
+				testContext := context.NewStandard()
+				standardNormalizer, err := normalizer.NewStandard(testContext)
+				Expect(err).To(BeNil())
+				bolusCalculator.Normalize(standardNormalizer)
+				Expect(bolusCalculator.Units).To(Equal(&common.MmolL))
+				Expect(bolusCalculator.BloodGlucoseInput).To(Equal(&expected))
+				Expect(bolusCalculator.InsulinSensitivity).To(Equal(&expected))
+				Expect(bolusCalculator.BloodGlucoseTarget.Target).To(Equal(&expected))
+			},
+				Entry("expected lower bg value", 3.7, 3.7),
```

Comment:
might not be the best way but I was just trying to go for the boundaries for these tests